### PR TITLE
feat(store): support non membership proof

### PR DESCRIPTION
**Pull request by rnbguy**
- Fix typo in `empty_child`
- Fix hashing with correct`empty_child`
- Extends `get_proof_rec` method to support non-membership proofs.
- Adds tests
- Fixes compilation for updated return signature

Note that, non-membership proofs require empty nodes. So, it wouldn't work with `GrowingStore` directly - as it should treat `[]` as an empty value.

### CODE REVIEW

File: crates/store/src/avl/proof.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ pub fn get_proof_spec() -> ProofSpec {
             child_size: 32,
             min_prefix_length: 0,
             max_prefix_length: 64,
-            empty_child: vec![0, 32],
+            empty_child: vec![0; 32],
```
Comment (rnbguy) [inline crates/store/src/avl/proof.rs:24]: we should use an `EMPTY_CHILD` constant and use it everywhere else.

File: crates/store/src/avl/tree.rs [hunk 2]
```diff
@@ -73,74 +74,151 @@ impl<K: Ord + AsBytes, V: Borrow<[u8]>> AvlTree<K, V> {
...
-        Some(CommitmentProof {
-            proof: Some(Proof::Exist(proof)),
-        })
+        let proof = Self::get_proof_rec(key, &self.root);
+        CommitmentProof { proof: Some(proof) }
     }
 
     /// Recursively build a proof of existence for the desired value.
-    fn get_proof_rec<Q: ?Sized>(key: &Q, node: &NodeRef<K, V>) -> Option<ExistenceProof>
+    fn get_proof_rec<Q: ?Sized>(key: &Q, node: &NodeRef<K, V>) -> Proof
```
Comment (rnbguy) [inline crates/store/src/avl/tree.rs:95]: The new implementation may need some helper functions to reuse repetitive code.

File: crates/store/src/avl/proof.rs (hunk 1 above)
Comment (rnbguy) [reply crates/store/src/avl/proof.rs]: a928605

**Review by ancazamfir — APPROVED** [review]
Very nice! :rocket:

File: basecoin/store/src/avl/tree.rs [hunk 3]
```diff
@@ -70,70 +69,110 @@ impl<K: Ord + AsBytes, V: Borrow<[u8]>> AvlTree<K, V> {
     }
 
     /// Return an existence proof for the given element, if it exists.
```
[the remark this answers is not in the document]
Comment (rnbguy) [reply basecoin/store/src/avl/tree.rs:71]: ca4751b

**Pull request merged**